### PR TITLE
Update luatest to 0.4.0, fix global labels verification error

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Update luatest version to 0.4.0 in CI.

Minor test changes: use new test hooks, use snake case in asserts, improve custom `ensure_throws` with luatest embedded `assert_error_msg_contains`, reorder metrics cleanup.

Fix global labels verification error.

### CODE REVIEW

File: test/collectors_test.lua [hunk 1]
```diff
@@ -6,29 +6,29 @@ local g = t.group('collectors')
...
-end
+g.after_each( function()
+    -- Delete all collectors and global labels
+    metrics.clear()
+end)
 
 g.test_counter = function()
-    ensure_throws('metrics.counter() w/o parameters', function()
+    -- metrics.counter() w/o parameters
+    t.assert_error( function()
```
Comment (sharonovd) [inline test/collectors_test.lua]: assert_error_msg_contains is usually self-explanatory

Comment (DifferentialOrange) [reply test/collectors_test.lua]: Replaced `assert_error` with `assert_error_msg_contains`. It also helped me to find the bug in the global labels verification error, thanks! (Fixed it in another commit of this MR.)

File: test/collectors_test.lua [hunk 2]
```diff
@@ -40,29 +38,30 @@ g.test_counter = function()
...
-    t.assertEquals(c:collect()[1].value, 8, '3 + 5 = 8')
+    t.assert_equals(c:collect()[1].value, 8, '3 + 5 = 8')
 
-    ensure_throws('counter:inc(<negative value>)', function()
+    t.assert_error_msg_contains("Counter increment should not be negative", function()
         c:inc(-1)
     end)
-    ensure_throws('counter:dec(...)', function()
+
+    t.assert_error_msg_contains("attempt to call method 'dec' (a nil value)", function()
```
Comment (olegrok) [inline test/collectors_test.lua]: counter.dec == nil should be enough. Why do you call it?

File: test/collectors_test.lua [hunk 3]
```diff
@@ -6,29 +6,27 @@ local g = t.group('collectors')
 local metrics = require('metrics')
 local utils = require('test.utils')
 
-g.before_all = function()
+g.before_all( function()
```
Comment (olegrok) [inline test/collectors_test.lua]: nil: space here is redundant: `g.before_all( function()` -> `g.before_all(function()`

File: test/collectors_test.lua (hunk 2 above)
Comment (DifferentialOrange) [reply test/collectors_test.lua]: Originally there was 
```
ensure_throws('counter:dec(...)', function()
        c:dec(1)
end)
```
and I simply rewrote it. Replaced with `assert_equals(..., nil, ...)`.

File: test/collectors_test.lua (hunk 3 above)
Comment (DifferentialOrange) [reply test/collectors_test.lua]: Removed redundant spaces in `before_all`/`after_each` calls.

**Pull request merged**